Source/JavaScriptCore/wasm/WasmSlowPaths.cpp

```diff
@@ -715,6 +649,103 @@ WASM_SLOW_PATH_DECL(call_ref_no_tls)
     return doWasmCallRef(callFrame, instance, reference, instruction.m_typeIndex);
 }
 
+static size_t jsrSize()
+{
+    static size_t jsrSize = 0;
+    static std::once_flag onceKey;
+    std::call_once(onceKey, [&] {
+        jsrSize = Wasm::wasmCallingConvention().jsrArgs.size();
+    });
+    return jsrSize;
+}
+
+WASM_SLOW_PATH_DECL(call_builtin)
+{
+    auto instruction = pc->as<WasmCallBuiltin>();
+    Register* stackBottom = callFrame->registers() - instruction.m_stackOffset;
+    Register* stackStart = stackBottom + CallFrame::headerSizeInRegisters + /* indirect call target */ 1;
+    Register* gprStart = stackStart + instruction.m_numberOfStackArgs;
+    Register* fprStart = gprStart + jsrSize();
+    UNUSED_PARAM(fprStart);
+
+    Wasm::LLIntBuiltin builtin = static_cast<Wasm::LLIntBuiltin>(instruction.m_builtinIndex);
+
+    unsigned gprIndex = 0;
+    unsigned fprIndex = 0;
+    unsigned stackIndex = 0;
+    auto takeGPR = [&]() -> Register& {
+        if (gprIndex != jsrSize()) {
+            unsigned cursor = gprIndex++;
+            return *(gprStart + cursor);
```

Comment:
Why not just express this as `gprStart[gprIndex++]`?

Reply:
Changed.